synapse/_scripts/synapse_port_db.py

```diff
@@ -460,7 +500,7 @@ async def handle_search_table(
 
         while True:
 
-            def r(txn):
+            def r(txn: LoggingTransaction) -> Tuple[List[str], List[Tuple]]:
```

Comment:
Laziness on my part. `Tuple` with no parameters is shorthand for `Tuple[Any, ...]`, i.e. a variadic tuple with no constraints on its elements' types. I'm using these in functions which work across multiple tables, so there's not an easy way to know the specific kind of tuple to expect.

It might be possible to do something better with sqlalchemy. There is a mypy plugin for use with sqlalchemy ORM, which I don't think we're interested in using for now. My reading of https://github.com/sqlalchemy/sqlalchemy/issues/6102#issuecomment-803230288 suggests that it'd be possible in principle to have typechecking that works on sqlachemy Core machinery; but that would need [typechecker support for variadic generics](https://github.com/python/mypy/issues/12280). I'm not fully sure how that fits into sqlalchemy's 1.4->2.0 transition.

But anyway, the above is aspirational; I don't see us changing our database machinery any time soon.

Reply:
> I'm using these in functions which work across multiple tables, so there's not an easy way to know the specific kind of tuple to expect.

This is what I was getting at. 👍